nilearn/plotting/html_stat_map.py

```diff
@@ -1,157 +1,496 @@
+                                percentile_func=fast_abs_percentile,
+                                name='threshold')
+
+    # Mask data
+    if threshold == 0:
+        data = np.ma.masked_equal(data, 0, copy=False)
+    else:
+        data = np.ma.masked_inside(data, -threshold, threshold, copy=False)
+    return data, threshold
+
+
+def _save_sprite(data, output_sprite, vmax, vmin, mask=None, cmap='Greys',
+                 format='png'):
+    """ Generate a sprite from a 3D Niimg-like object.
+    """
+
+    # Create sprite
+    sprite = _data2sprite(data)
+
+    # Mask the sprite
+    if mask is not None:
+        mask = _data2sprite(mask)
+        sprite = np.ma.array(sprite, mask=mask)
+
+    # Save the sprite
+    imsave(output_sprite, sprite, vmin=vmin, vmax=vmax, cmap=cmap,
+           format=format)
+
+    return sprite
+
+
+def _bytesIO_to_base64(handle_io):
+    """ Encode the content of a bytesIO virtual file as base64.
+        Also closes the file.
+    """
+    handle_io.seek(0)
+    data = encodebytes(handle_io.read()).decode('utf-8')
+    handle_io.close()
+    return data
+
+
+def _save_cm(output_cmap, cmap, format='png', n_colors=256):
+    """ Save the colormap of an image as an image file.
+    """
+
+    # save the colormap
+    data = np.arange(0., n_colors) / (n_colors - 1.)
+    data = data.reshape([1, n_colors])
+    imsave(output_cmap, data, cmap=cmap, format=format)
+
+
+def _sanitize_cm(cmap, n_colors=256):
+    """Make sure that the colormap has no duplicated colors.
+    """
+    # Extract a list of colors
+    cmap = plt.cm.get_cmap(cmap)
+    cmaplist = [cmap(i) for i in range(cmap.N)]
+
+    # Filter out any duplicated colors
+    mask = [cmaplist.count(cmap(i)) == 1 for i in range(cmap.N)]
+    if mask.count(True) < 2:
+        raise ValueError('The selected colormap has too many '
+                         'duplicated colors')
+    cmaplist = np.array(cmaplist)[np.array(mask)]
+
+    # Return a matplotlib colormap with the original number of colors
+    return LinearSegmentedColormap.from_list('Custom cmap', cmaplist, n_colors)
 
 
 class StatMapView(HTMLDocument):
     pass
 
 
-def _to_papaya(cmap, norm, abs_min, vmax, symmetric_cmap):
-    """Create colormaps ('lookup tables') given to papaya viewer"""
-    n_colors = 100
-    if not symmetric_cmap:
-        x = np.linspace(norm(abs_min), norm(vmax), n_colors)
-        rgb = cmap(x)[:, :3]
-        return {'positive_cmap': _rgb_to_papaya(x, rgb),
-                'negative_cmap': None}
-    papaya = {}
-    x = np.linspace(norm(-vmax), norm(-abs_min), n_colors)
-    y = np.linspace(0, 1, n_colors)
-    rgb = cmap(x)[::-1, :3]
-    papaya['negative_cmap'] = _rgb_to_papaya(y, rgb)
-    x = np.linspace(norm(abs_min), norm(vmax), n_colors)
-    rgb = cmap(x)[:, :3]
-    papaya['positive_cmap'] = _rgb_to_papaya(y, rgb)
-    return papaya
+def _mask_stat_map(stat_map_img, threshold=None):
+    """ Load a stat map and apply a threshold.
+    """
```

Comment:
I have not written any doc string for private functions. I believe that's consistent with the rest of the code base (at least what I looked at). Let me know if you would like doc strings for those.

Reply:
Private functions should also have proper docstrings. Whatever the reason other private functions in this library do not have docstrings (they should), in this one, multiple outputs are being returned. It would be helpful not to have to look at the code when using it in future to figure out whats going on. Listing the variable names should be sufficient, they are sufficiently self-describing.

Reply:
added lists of returned variables for all private functions.